### PR TITLE
Add contributing guide

### PR DESCRIPTION
**Pull request by AlekseySh**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+## Before you start
+* Read our [FAQ](https://github.com/OML-Team/open-metric-learning#faq).
+* Check out [python examples](https://github.com/OML-Team/open-metric-learning#get-started-using-python)
+  and [config examples](https://github.com/OML-Team/open-metric-learning/tree/main/examples).
+
+## Choosing a task
+* You can work on one of the [existing issues](https://github.com/OML-Team/open-metric-learning)
```
Comment (DaloroAT) [inline CONTRIBUTING.md:7]: url

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  or create the new one. Especially pay attention to the issues marked with the `good_first_issue` flag.
+* Start the conversation under the issue that you picked. We will discuss the design and content of the pull request and
+  then you can start working on it.
+
+## Contributing
+* Fork the repository.
+* Clone it locally.
+* Create a branch with a name that speaks for itself.
+* Set up the environment. You can install the library in dev mode via `pip install -e .`
+  or [build / pull docker image](https://github.com/OML-Team/open-metric-learning#installation).
```
Comment (DaloroAT) [inline CONTRIBUTING.md:17]: We need to actualize docker before suggesting developing in this way

File: CONTRIBUTING.md (hunk 1 above)
Comment (AlekseySh) [reply CONTRIBUTING.md:7]: thx

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+* Implement the discussed functionality, **docstrings**, and **tests** for it.
+* Run tests locally via `make run_tests` or `make docker_tests` (if you use docker).
```
Comment (DaloroAT) [inline CONTRIBUTING.md:19]: always use `make docker_tests`. Probably we don't want to manage locals envs of users

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+* Push the code to your forked repository.
+* Create a pull request to OpenMetricLearning.
+
+## Good to know
```
Comment (b0nce) [inline CONTRIBUTING.md:23]: Please add info about configs for each new model/loss/...

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
+## Before you start
+* Read our [FAQ](https://github.com/OML-Team/open-metric-learning#faq).
+* Check out [python examples](https://github.com/OML-Team/open-metric-learning#get-started-using-python)
+  and [config examples](https://github.com/OML-Team/open-metric-learning/tree/main/examples).
+
+## Choosing a task
+* You can work on one of the [existing issues](https://github.com/OML-Team/open-metric-learning)
+  or create the new one. Especially pay attention to the issues marked with the `good_first_issue` flag.
+* Start the conversation under the issue that you picked. We will discuss the design and content of the pull request and
```
Comment (b0nce) [inline CONTRIBUTING.md:9]: Comma before "and".

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+* Create a branch with a name that speaks for itself.
+* Set up the environment. You can install the library in dev mode via `pip install -e .`
+  or [build / pull docker image](https://github.com/OML-Team/open-metric-learning#installation).
+* Implement the discussed functionality, **docstrings**, and **tests** for it.
+* Run tests locally via `make run_tests` or `make docker_tests` (if you use docker).
+* Push the code to your forked repository.
+* Create a pull request to OpenMetricLearning.
+
+## Good to know
+* If your want to change `README.md` you should go to `docs/readme`, change the desired section and then build
```
Comment (b0nce) [inline CONTRIBUTING.md:24]: "If you" instead of "if your".

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+  readme via `make build_readme`. *So, don't change the main readme file directly, otherwise tests will fail.*
+* Don't forget to update the documentation if needed. Its source is located in `docs/source`. To inspect
+  it locally you should run `make html` (from `docs` folder) and then open `docs/build/html/index.html` in your
```
Comment (b0nce) [inline CONTRIBUTING.md:27]: Comma after "locally".

File: CONTRIBUTING.md (hunk 4 above)
Comment (AlekseySh) [reply CONTRIBUTING.md:23]: okay

File: CONTRIBUTING.md (hunk 5 above)
Comment (AlekseySh) [reply CONTRIBUTING.md:9]: don't need

File: CONTRIBUTING.md (hunk 6 above)
Comment (AlekseySh) [reply CONTRIBUTING.md:24]: thx

File: CONTRIBUTING.md (hunk 2 above)
Comment (AlekseySh) [reply CONTRIBUTING.md:17]: discussed offline

File: CONTRIBUTING.md (hunk 3 above)
Comment (AlekseySh) [reply CONTRIBUTING.md:19]: discussed

**Review by b0nce — APPROVED** [review]

**Pull request merged**